src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -819,8 +819,16 @@ private void AddEasyHandle(RequestCompletionSource state)
                         throw new HttpRequestException(SR.net_http_client_execution_error,
                             GetCurlException(result, true));
                     }
+                    state.SessionHandle = _multiHandle;
+                    _multiHandle.FdCount = _multiHandle.FdCount + 1;
+                    if (_multiHandle.PollCancelled)
+                    {
+                        // TODO: Create single polling thread for all HttpClientHandler objects
+                        Task.Factory.StartNew(s => PollFunction(((CurlHandler)s)._multiHandle), this,
+                                CancellationToken.None, TaskCreationOptions.DenyChildAttach, TaskScheduler.Default);
+                        _multiHandle.PollCancelled = false;
```

Comment:
I suggest moving this ```_multiHandle.PollCanceled = false``` line to before you start the task.  I realize there aren't any race conditions around it currently because all access to PollCancelled is done inside of the lock that we're currently in.  Still seems like a good measure.